src/pykeen/models/base.py

```diff
@@ -1032,166 +1093,455 @@ def load_state(self, path: str) -> None:
     ) -> None:
-        """Initialize the entity embedding model.
-
-        :param embedding_dim:
-            The embedding dimensionality. Exact usages depends on the specific model subclass.
+        """Initialize the module.
 
-        .. seealso:: Constructor of the base class :class:`pykeen.models.Model`
+        :param triples_factory:
+            The triples factory facilitates access to the dataset.
+        :param loss:
+            The loss to use. If None is given, use the loss default specific to the model subclass.
+        :param predict_with_sigmoid:
+            Whether to apply sigmoid onto the scores when predicting scores. Applying sigmoid at prediction time may
+            lead to exactly equal scores for certain triples with very high, or very low score. When not trained with
+            applying sigmoid (or using BCEWithLogitsLoss), the scores are not calibrated to perform well with sigmoid.
+        :param automatic_memory_optimization:
+            If set to `True`, the model derives the maximum possible batch sizes for the scoring of triples during
+            evaluation and also training (if no batch size was given). This allows to fully utilize the hardware at hand
+            and achieves the fastest calculations possible.
+        :param preferred_device:
+            The preferred device for model training and inference.
+        :param random_seed:
+            A random seed to use for initialising the model's weights. **Should** be set when aiming at reproducibility.
         """
         super().__init__(
             triples_factory=triples_factory,
             automatic_memory_optimization=automatic_memory_optimization,
             loss=loss,
             preferred_device=preferred_device,
             random_seed=random_seed,
-            regularizer=regularizer,
             predict_with_sigmoid=predict_with_sigmoid,
         )
-        self.entity_embeddings = Embedding.init_with_device(
-            num_embeddings=triples_factory.num_entities,
-            embedding_dim=embedding_dim,
-            device=self.device,
-            initializer=entity_initializer,
-            initializer_kwargs=entity_initializer_kwargs,
-            normalizer=entity_normalizer,
-            normalizer_kwargs=entity_normalizer_kwargs,
-            constrainer=entity_constrainer,
-            constrainer_kwargs=entity_constrainer_kwargs,
+        self.entity_representations = _prepare_representation_module_list(representations=entity_representations)
+        self.relation_representations = _prepare_representation_module_list(representations=relation_representations)
+        self.interaction = interaction
+        # Comment: it is important that the regularizers are stored in a module list, in order to appear in
+        # model.modules(). Thereby, we can collect them automatically.
+        self.weight_regularizers = nn.ModuleList()
+
+    def append_weight_regularizer(
```

Comment:
@mberr this function is only used during tests. Can it be moved out of the base class?

Reply:
No, because it is the proper/public way to add weight regularizer's.